Further improvements to configuration documentation (#429)
NVIDIA-NeMo/Gym: docs/about/concepts/configuration.md

# Configuration System

NeMo Gym uses YAML configuration files to define [Model, Resources, and Agent servers](./core-abstractions). Each server gets its own configuration block, providing modular control over the entire training environment.

## How Servers Connect

> A training environment typically includes all three server types working together. The Agent server config specifies which Model and Resources servers to use by referencing their server IDs. This wiring is what ties each training environment together — the Agent knows which Model to call and which Resources to use.

Reviewer: ```suggestion
A training environment typically includes all three server types working together. The Agent server config specifies which Model and Resources servers to use by referencing their server IDs. These references connect each training environment together — the Agent knows which Model to call and which Resources to use.
```
